src/sfctl/custom_app.py

```diff
@@ -144,14 +172,37 @@ def upload(path, imagestore_string='fabric:ImageStore', show_progress=False):  #
     # Upload to either to a folder, or native image store only
     if 'file:' in imagestore_string:
         dest_path = path_from_imagestore_string(imagestore_string)
-        upload_to_fileshare(abspath, os.path.join(dest_path, basename),
-                            show_progress)
+
+        process = Process(target=upload_to_fileshare,
```

Comment:
If the user cancels the command while the process is still running, what happens?

Reply:
They will have to re-upload.